src/pre-engagement-form/localization.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import type { Manager } from '@twilio/flex-webchat-ui';
+import PubSub from 'pubsub-js';
+
+type LocalizationProviderProps = {
+  manager: Manager;
+  children: JSX.Element;
+};
+
+const LocalizationContext = createContext({});
+const STRINGS_UPDATED = 'STRINGS_UPDATED';
+
+export const notifyStringsUpdated = () => PubSub.publish(STRINGS_UPDATED);
+
+export const LocalizationProvider: React.FC<LocalizationProviderProps> = ({ manager, children }) => {
+  const [strings, setStrings] = useState(manager.strings);
+
+  useEffect(() => {
+    const token = PubSub.subscribe(STRINGS_UPDATED, () => setStrings(manager.strings));
+
+    return () => {
+      PubSub.unsubscribe(token);
+    };
```

Comment:
We only need to unsubscribe to the last `token`? Cause this will run only when the component unmounts :thinking:

Reply:
Actually, the cleanup function will be run according to the **dependency array** as well (not only when it unmounts).
So, in practice, we're unsubscribing every token.

Reply:
We actually need it in the dependency array especially if `manager` changes. Because use `manager.strings` as the source and we need to cleanup subscriptions in case manager changes.

(We have comments below on managing strings in "our own redux", that would be possibly the preferred way to go)